controllers/openstackmachine_controller.go

```diff
@@ -334,8 +334,6 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 
 	// Set an error message if we couldn't find the instance.
 	if instanceStatus == nil {
-		err = errors.New("OpenStack instance not found")
```

Comment:
I wonder if the `if instanceStatus == nil` check here is bogus. Looking at https://github.com/Nordix/cluster-api-provider-openstack/blob/3c742d2cd136ca2c64b1245a97230ae038eafe84/controllers/openstackmachine_controller.go#L427-L444 I don't think this should ever happen. One of the following things should have happened:
* The instance existed and we returned it (instanceStatus != nil)
* The instance did not exist and we created it successfully (instanceStatus != nil)
* The there was an error either determining if the instance exists, or creating the instance (err != nil, handled above)

I think this might actually be dead code. Probably best to delete it entirely. I think this code is my fault, actually. I think I was intending to make additional changes in which CreateInstance could return nil but no error. Looking at it now I don't think there are any. Lets not leave in half-implemented code.

Are you sure this is the code which is causing your issue?

Reply:
Oh I see! I was somehow mixing this up with the same check in `getOrCreate` below :facepalm: 
I will remove it